### ISSUE TITLE
[results] How to normalize AUCell output?

### ISSUE THREAD

**Issue by BanafshehKhaki** [aertslab/pySCENIC#376]
Hello,

I really enjoy exploring/using pyscenic. Thank you for the great work! And I appreciate your help in figuring out my issue below:

I want to compare the AUC results (produced by AUCell) between different samples (run statistics on auc_mtx between different patients' samples). However, few points I have understood so far:

1. AUC outputs are not normalized and hence "can only be used to compare the activity of a regular across cells of the same experiment.." (Scenic protocol, p.2257) I understand this means I can't use AUC values directly for my purpose?

2. The aucell function in aucell.py has a normalization function to normalize AUC values to maximum 1 , and it is set to False by default: "normalize: bool = False," (from GitHub: master/src/pyscenic/aucell.py)
3. The pyscenic aucell command does not have any normalization parameter and hence I cannot change the 'normalize' variable in the aucell.py.  Or can we in a specific way? Or is there a reason that it's not included in the parameters list ?

4. However, the pyscenic aucell command has '--nes_threshold' parameter, So I am assuming NES is calculated (auc-auc.mean/ auc.std), am I right? Can we get nes_mtx for cells X regulons?  
5- Or is this (--nes_threshold) for retrieving regulons from ctx (I-cisTarget) output ?

5- I see that in R packages the normalize parameters is normAUC and by default is set to True. am I correct ?(Github: R/02_calcAUC.R)

In short, I appreciate your suggestions for comparing AUCell results (auc matrices) across different samples? 

Thank you,

**Comment by SeppeDeWinter** [comment]
Hi @BanafshehKhaki 

1. That's correct
2 + 3. That's also correct. As far as I'm aware there is no particular reason why this is not included as a parameter.
4. The NES threshold relates to the cistarget output, like you say. So it has nothing to do with normalising the AUC values.
5. That's correct

If you want to normalise your AUC values you have two options:
1. Recalculate the AUC values inside a python kernel (i.e. using a custom script) setting the normalise parameter to True
2. Normalising the AUC values afterwards. The normalisation used in the package is quite simple, it just divides the AUC values of each regulon by the maximum of the regulon's AUC value. You can choose to do some more complicated normalisation (e.g. Z-score, ...)

As to what is the best way to normalise and compare these values I can not really give advise. This will be very specific to the question you are asking. 

Hope this helps.

Best,

S.
